src/Microsoft.DotNet.TemplateLocator/TemplateLocator.cs

```diff
@@ -5,68 +5,106 @@
 using System.Collections.Generic;
 using System.IO;
 using System.Linq;
-using NuGet.Common;
-using NuGet.Protocol;
+using Microsoft.DotNet.DotNetSdkResolver;
+using Microsoft.NET.Sdk.WorkloadManifestReader;
 
 namespace Microsoft.DotNet.TemplateLocator
 {
     public sealed class TemplateLocator
     {
-        private DirectoryInfo _dotnetSdkTemplatesLocation;
+        private readonly Lazy<NETCoreSdkResolver> _netCoreSdkResolver;
 
         public TemplateLocator()
+            : this(Environment.GetEnvironmentVariable, VSSettings.Ambient)
         {
-            string mockTemplateLocation = Environment.GetEnvironmentVariable("MOCKDOTNETSDKTEMPLATESLOCATION");
-            if (!string.IsNullOrWhiteSpace(mockTemplateLocation))
-            {
-                _dotnetSdkTemplatesLocation = new DirectoryInfo(mockTemplateLocation);
-            }
         }
 
-        public IReadOnlyCollection<IOptionalSdkTemplatePackageInfo> GetDotnetSdkTemplatePackages(string sdkVersion)
+        // Test constructor
+        public TemplateLocator(Func<string, string> getEnvironmentVariable, VSSettings vsSettings)
+        {
+            _netCoreSdkResolver =
+                new Lazy<NETCoreSdkResolver>(() => new NETCoreSdkResolver(getEnvironmentVariable, vsSettings));
+        }
+
+        public IReadOnlyCollection<IOptionalSdkTemplatePackageInfo> GetDotnetSdkTemplatePackages(
+            string sdkVersion,
+            string dotnetRootPath)
         {
-            if (_dotnetSdkTemplatesLocation == null)
+            var sdkVersionParsed = Version.Parse(sdkVersion.Split('-')[0]);
+
+            bool FindSameVersionBand(DirectoryInfo manifestDirectory)
             {
-                return Array.Empty<IOptionalSdkTemplatePackageInfo>();
+                var directoryVersion = Version.Parse(manifestDirectory.Name.Split('-')[0]);
+                return directoryVersion.Major == sdkVersionParsed.Major &&
+                       directoryVersion.Minor == sdkVersionParsed.Minor &&
+                       (directoryVersion.Revision / 100) == (sdkVersionParsed.Revision / 100);
             }
 
-            IEnumerable<LocalPackageInfo> packages = LocalFolderUtility
-                            .GetPackagesV2(_dotnetSdkTemplatesLocation.FullName, new NullLogger());
+            var bondManifestDirectory = new DirectoryInfo(dotnetRootPath).GetDirectories().Where(FindSameVersionBand)
```

Comment:
miss the manfiest folder directory. Workloadmanifest directory

Reply:
`dotnet/sdk/{sdk-band}-manifests/{manifest-id}/`

At least missing `sdk` folder. But there might be a different spec

Reply:
use <DOTNET ROOT>/workloadmanifests/<FEATURE VERSION BAND>/